src/libraries/Common/tests/Tests/System/StringTests.cs

```diff
@@ -3194,7 +3197,7 @@ public static void IndexOf_TurkishI_EnglishUSCulture()
             }
         }
 
-        [ConditionalFact(typeof(PlatformDetection), nameof(PlatformDetection.IsNotInvariantGlobalization))]
+        [ConditionalFact(typeof(PlatformDetection), nameof(PlatformDetection.IsIcuGlobalizationAndNotHybrid))]
```

Comment:
As `IsIcuGlobalizationAndNotHybrid => IsIcuGlobalization && IsNotHybridGlobalization` .
Either `IsInvariantGlobalization` will be true, either`IsIcuGlobalization` will be true, both same time can't. With that logic test will not run when `Invariant` or `Hybrid` mode.

Reply:
It still feels like this change will skip the test for some situations. For example, the test was running for `IsHybridGlobalizationOnBrowser` before, but now with `IsNotHybridGlobalization`, it will be skipped. `IsNlsGlobalization` will also now be skipped as it is `IsNotInvariantGlobalization && !IsIcuGlobalization`.

Reply:
yes that's true, updated it.